api-reference/docstubs/api/calendar-allowedcalendarsharingroles.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "allowedCalendarSharingRoles"
+description: ""
+author: ""
+localization_priority: Normal
+ms.prod: ""
+doc_type: apiPageType
+---
+
+# allowedCalendarSharingRoles
+
+Namespace: microsoft.graph
+
+
+
+## Permissions
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/concepts/permissions-reference.md).
+
+|Permission type|Permissions (from most to least privileged)|
+|:---|:---|
+|Delegated (work or school account)|**TODO: Determine scopes **|
+|Delegated (personal Microsoft account)|Not supported.|
+|Application|**TODO: Determine AppOnly scopes **|
+
+## HTTP request
+<!-- {
+  "blockType": "ignored"
+}
+-->
+``` http
+GET /me/calendar/allowedCalendarSharingRoles
+GET /users/{usersId}/calendar/allowedCalendarSharingRoles
+GET /me/calendars/{calendarId}/allowedCalendarSharingRoles
+GET /groups/{groupsId}/calendar/allowedCalendarSharingRoles
+GET /me/joinedGroups/{groupId}/calendar/allowedCalendarSharingRoles
+GET /me/messages/{messageId}/event/calendar/allowedCalendarSharingRoles
+GET /users/{usersId}/calendars/{calendarId}/allowedCalendarSharingRoles
+GET /me/calendarGroups/{calendarGroupId}/calendars/{calendarId}/allowedCalendarSharingRoles
+```
+
+## Request headers
+|Name|Description|
+|:---|:---|
+|Authorization|Bearer {token}.Required|
+
+## Request body
+In the request URL, provide the following query parameters with values.
+The following table shows the parameters that can be used with this function.
+
+|Property|Type|Description|
+|:---|:---|:---|
+|User|String||
```

Comment:
Add **TODO Description** if it cannot be found in metadata, for parameter descriptions.